refactor(compiler): add support for sanitizing properties and attributes

mmalerba opened the pull request:
Sets sanitizer functions when attempting to set sensitive properties and attributes

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
sanitizer functions are not supported in template pipeline

Issue Number: N/A


## What is the new behavior? 
sanitizer functions are supported in template pipeline

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

mmalerba commented on packages/compiler/src/template/pipeline/src/phases/resolve_sanitizers.ts:
```diff
@@ -0,0 +1,84 @@
+          // The compiler does not fill in a sanitizer for style binding values because the
+          // style algorithm knows internally what props are subject to sanitization (only style
+          // values set via [attr.style] are explicitly sanitized)
+          // TODO: is the context ever anything other than style for these props? if not we can just
```
Can we just completely remove sanitization for style prop/map?

dylhunn commented on packages/compiler/src/template/pipeline/src/phases/reify.ts:
```diff
@@ -7,11 +7,21 @@
 import type {ComponentCompilation, ViewCompilation} from '../compilation';
 import * as ng from '../instruction';
 
+// TODO: Ok to pull in `Identifiers` here? I mostly see it used in `instruction.ts`
```
I don't see why not? If there's a problem, @alxhub might know

dylhunn commented on packages/compiler/src/template/pipeline/src/phases/resolve_sanitizers.ts:
```diff
@@ -0,0 +1,84 @@
+/**
+ * Gets a map of all elements in the givne view by their xref id.
+ */
+// TODO: Is there a shared location I can put this? Copied the code from attribute_extraction.ts
```
Not yet, but feel free to create a shared.ts!

dylhunn reviewed (approved):
reviewed-for: fw-compiler

alfaproject commented on packages/compiler/src/template/pipeline/ir/src/ops/update.ts:
```diff
@@ -6,6 +6,15 @@
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
```
This header is duplicated now

mmalerba committed:
refactor(compiler): add support for sanitizing properties and attributes

Sets sanitizer functions when attempting to set sensitive properties and attributes

mmalerba commented:
Does not affect internal tests

alxhub commented:
This PR was merged into the repository by commit c1052cf7a77e0bf2a4ec14f9dd5abc92034cfd2e.
